app/src/main/java/com/wut/indoornavigation/logic/graph/impl/GraphImpl.java

```diff
@@ -0,0 +1,209 @@
+package com.wut.indoornavigation.logic.graph.impl;
+
+
+import android.support.annotation.NonNull;
+
+import com.wut.indoornavigation.logic.graph.Graph;
+import com.wut.indoornavigation.logic.graph.UnionFind;
+import com.wut.indoornavigation.logic.graph.models.Edge;
+import com.wut.indoornavigation.logic.graph.models.Vertex;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.PriorityQueue;
+
+
+public class GraphImpl implements Graph {
+    private final List<Vertex> vertices;
+    private final Map<Vertex, List<Edge>> edges;
+
+
+    public GraphImpl(@NonNull List<Vertex> vertices) {
+        this.vertices = new ArrayList<>();
+        this.edges = new HashMap<>();
+        this.vertices.addAll(vertices);
+    }
+
+    @Override
+    public boolean addEdge(@NonNull Edge edge) {
+        Vertex from = edge.getFrom();
+        Vertex to = edge.getTo();
+
+        if(vertices.contains(from) && vertices.contains(to)){
+            if(!edges.containsKey(from)){
+                List<Edge> outEdges = new ArrayList<>();
+                outEdges.add(edge);
+                edges.put(from, outEdges);
+
+                return true;
+            } else{
+                List<Edge> fromOutEdges = edges.get(from);
+                if(fromOutEdges.contains(edge)){
+                    return false;
+                }else{
+                    fromOutEdges.add(edge);
+
+                    return true;
+                }
+            }
+        }
+
+        return false;
+    }
+
+    @Override
+    public boolean addEdge(@NonNull Vertex start, @NonNull Vertex end, double weight) {
```

Comment:
to mi troche ulatwia sprawe, nie musze tworzyc obiektu tylko po to by wstawic krawędź, poza tym to tylko przeciążenie metody :P